### PR TITLE
Remove extra semicolons in some rules, because they now matter

### PR DESCRIPTION
**Pull request by aryx**
Because of https://github.com/returntocorp/semgrep/pull/2028
which is needed for https://github.com/returntocorp/semgrep/issues/1948
we do not remove anymore semicolons in expression statement patterns
(e.g., 'foo(...);')
As a consequence, if you have some semicolons in a pattern, because
we do not remove them, they have an influence on the matching process
(the range of the matched code will be 1 longer, and this may have
bad effect when combining pattern not using semicolon with pattern-not
patterns using semicolons).

Test plan:
use latest semgrep via pipenv shell and
$ make test

### CODE REVIEW

**Comment by aryx** [comment]
@ajinabraham this might have an impact in some of your rules also (even though I didn't see any regression
in our contrib/nodejsscan directory).
TL;DR: you should not use semicolons in patterns if what you want to match is an expression, e.g., use foo(...) not foo(...);

**Comment by ajinabraham** [comment]
Thanks for the heads up. Will watchout for the changes in 0.32.0

File: javascript/jquery/security/audit/jquery-insecure-selector.yaml [hunk 1]
```diff
@@ -49,7 +49,7 @@ rules:
             }
       - pattern-either:
         - pattern: |
-            $(<... $Y ...>);
+            $(<... $Y ...>)
         - pattern: |
             $("..." + (<... $Y ...>));
```
Comment (mschwager) [inline javascript/jquery/security/audit/jquery-insecure-selector.yaml]: Should we remove this semi-colon too then?

File: javascript/jquery/security/audit/jquery-insecure-selector.yaml [hunk 2]
```diff
@@ -49,7 +49,7 @@ rules:
...
         - pattern: |
```
Comment (mschwager) [inline javascript/jquery/security/audit/jquery-insecure-selector.yaml:55]: Same here.

**Review by mschwager — APPROVED** [review]
LGTM - we may be able to remove 2 more

File: javascript/jquery/security/audit/jquery-insecure-selector.yaml (hunk 1 above)
Comment (aryx) [reply javascript/jquery/security/audit/jquery-insecure-selector.yaml]: yep, I will remove them too

**Comment by ajinabraham** [comment]
@aryx Is there a regression in semgrep 0.35 regarding semicolons?
I made the changes to rules as suggested (removed semicolons) and everything worked till 0.34
and broken in 0.35

https://semgrep.dev/s/ajinabraham:rule_semgrep?version=0.34.0
https://semgrep.dev/s/ajinabraham:rule_semgrep?version=0.35.0
